web/js/modules/image-download/util.js

```diff
@@ -43,9 +85,23 @@ export function getLatestIntervalTime(layerDefs, dateTime) {
  */
 export function getDownloadUrl(url, proj, layerDefs, lonlats, dimensions, dateTime, fileType, isWorldfile, markerCoordinates) {
   const { crs } = proj.selected;
-  const layersArray = imageUtilGetLayers(layerDefs, proj.id);
-  const layerWraps = imageUtilGetLayerWrap(layerDefs);
-  const opacities = imageUtilGetLayerOpacities(layerDefs);
+  let layersArray = imageUtilGetLayers(layerDefs, proj.id);
```

Comment:
Since you are also checking for the presence of orbit track layers within the `addOrbitTrackRevisions` function, you could skip checking for it here and condense the changes to this function by quite a few lines to something like:
```
  const {
    layersArray,
    layerWraps,
    opacities,
  } = addOrbitTrackRevisions(
    imageUtilGetLayers(layerDefs, proj.id),
    imageUtilGetLayerWrap(layerDefs),
    imageUtilGetLayerOpacities(layerDefs),
  );
```

This would require reconciling the names of the properties being returned from `addOrbitTrackRevisions` with those used in the rest of this function.

Reply:
I like that! Will do.